### PR TITLE
bra-ket and plus/minus for Latex parser

### PR DESCRIPTION
**Pull request by bhpayne**
#### References to other Issues or PRs
https://github.com/sympy/sympy/issues/19854

#### Brief description of what is fixed or changed
@iammosespaulr  added tokens  plus/minus and for  bra-ket Dirac notation

#### Other comments
In collaboration with @msgoff and @iammosespaulr

#### Release Notes

* parsing
   * Improved Latex support for reading expressions into SymPy

### CODE REVIEW

File: sympy/parsing/latex/_parse_latex_antlr.py [hunk 1]
```diff
@@ -319,7 +319,13 @@ def convert_atom(atom):
     elif atom.mathit():
         text = rule2text(atom.mathit().mathit_text())
         return sympy.Symbol(text)
-
+    elif atom.bra():
```
Comment (iammosespaulr) [inline sympy/parsing/latex/_parse_latex_antlr.py:322]: you gotta add this import at the top 
```python3
from sympy.physics.quantum.state import Bra, Ket
```

File: sympy/parsing/tests/test_latex.py [hunk 2]
```diff
@@ -143,6 +144,8 @@ def test_import():
      Integral(3*_Pow(theta, -1), theta)),
     ("\\int \\frac{1}{x} + 1 dx", Integral(_Add(_Pow(x, -1), 1), x)),
     ("x_0", Symbol('x_{0}')),
+    ("\langle x |", Bra('x')),
```
Comment (iammosespaulr) [inline sympy/parsing/tests/test_latex.py:147]:
```suggestion
    ("\\langle x |", Bra('x')),
```

File: sympy/parsing/tests/test_latex.py [hunk 3]
```diff
@@ -143,6 +144,8 @@ def test_import():
...
+    ("| x \rangle", Ket('x')),
```
Comment (iammosespaulr) [inline sympy/parsing/tests/test_latex.py:148]:
```suggestion
    ("| x \\rangle", Ket('x')),
```

**Comment by bhpayne** [comment]
fixed the issues that @iammosespaulr pointed out

**Comment by czgdp1807** [comment]
@iammosespaulr Is it good to go?

**Comment by czgdp1807** [comment]
@bhpayne Could you please resolve the conflicts if you are interested in continuing this? 
@iammosespaulr @asmeurer Please review.

**Pull request closed without merging**